### PR TITLE
An effect part of the condeff should be AndEffect and not And.

### PR DESCRIPTION
**Pull request by haz**
Minor fix for #118

### CODE REVIEW

**Comment by haz** [comment]
Probably best to wait for #120 to be resolved before this is merged (so we can be sure nothing's broken).

File: tests/fixtures/code_objects/blocksworld_fond.py
Comment (haz) [inline tests/fixtures/code_objects/blocksworld_fond.py:63]: This is probably the biggest impact of this change...using the shorthand `&` between fluents will automatically use the `And`, but if it's an effect, then it should be an `AndEffect` (I think?)

File: tests/fixtures/code_objects/blocksworld_fond.py
Comment (haz) [inline tests/fixtures/code_objects/blocksworld_fond.py:63]: This is probably the biggest impact of this change...using the shorthand `&` between fluents will automatically use the `And`, but if it's an effect, then it should be an `AndEffect` (I think?)

File: tests/test_formatter.py
Comment (haz) [inline tests/test_formatter.py:139]: We ok with this? Comes from the fact that `And` has `idempotency = True`

File: pddl/logic/effects.py
Comment (haz) [inline pddl/logic/effects.py:130]: This weaken things a bit, but I couldn't figure out how to optionally get the type in `And`.

File: pddl/logic/effects.py
Comment (haz) [inline pddl/logic/effects.py:27]: We lose `functools.total_ordering` in this shift. That ok?

File: tests/test_formatter.py
Comment (francescofuggitti) [reply tests/test_formatter.py:139]: I'm ok with it.

File: pddl/logic/effects.py
Comment (francescofuggitti) [reply pddl/logic/effects.py:128]: Yeah...why can't we apply the optional to the `And` here? (Unfortunately, I cannot attempt any trial as I don't have access to my laptop at the moment)

File: pddl/logic/effects.py
Comment (francescofuggitti) [reply pddl/logic/effects.py:27]: This should be ok as we get `__eq__` from the `BinaryOp,` but it might be a problem if we need an ordering of operands like `__lt__.`

**Review by marcofavorito — APPROVED** [review]

File: pddl/logic/effects.py
Comment (haz) [reply pddl/logic/effects.py:27]: Ok. Until we need it, fine without it ;).

File: pddl/logic/effects.py
Comment (haz) [reply pddl/logic/effects.py:128]: It's because the `AndEffect` inherited from `Generic` with a custom effect type: https://github.com/AI-Planning/pddl/blob/main/pddl/logic/effects.py#L28

`And` does not, though, and I couldn't see a clear path to making it work.

Comment (haz) [reply pddl/logic/effects.py:128]: Going to resolve and merge, but happy to revisit if it becomes necessary. Ultimately what I think this will allow through is the parsing of `(and (p) (or (q) (s)))`, as an effect, which should ideally be an error. That syntax is ok as a precondition (or condition in a conditional effect), but not as an effect.

**Pull request merged**
